### PR TITLE
[DROOLS-6646] move ClassFieldAccessorStore out of drools-core

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6646

Related PRs
https://github.com/kiegroup/drools/pull/4116
https://github.com/kiegroup/kogito-runtimes/pull/1862

### CODE REVIEW

File: drools-traits/src/main/resources/META-INF/kie/drools-traits/kie.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 org.drools.compiler.builder.impl.TypeDeclarationBuilderFactory=org.drools.traits.compiler.builder.impl.TraitTypeDeclarationBuilderFactory
 ?org.kie.internal.process.CorrelationKeyFactory = org.jbpm.persistence.correlation.JPACorrelationKeyFactory
 org.drools.core.reteoo.RuntimeComponentFactory=org.drools.traits.core.reteoo.TraitRuntimeComponentFactoryImpl;1
-org.drools.core.reteoo.CoreComponentFactory=org.drools.traits.core.reteoo.TraitCoreComponentFactory
+org.drools.core.reteoo.CoreComponentFactory=org.drools.traits.core.reteoo.TraitCoreComponentFactory;1
```
Comment (danielezonca) [inline drools-traits/src/main/resources/META-INF/kie/drools-traits/kie.conf:4]: Why this change? Do we have a new CoreComponentFactory impl?

File: drools-core/src/main/java/org/drools/core/base/AccessorKey.java [hunk 2]
```diff
@@ -109,4 +108,32 @@ public String toString() {
...
+     * Utility method to take a string and convert it to normal Java variable
+     * name capitalization.  This normally means converting the first
+     * character from upper case to lower case, but in the (unusual) special
+     * case when there is more than one character and both the first and
+     * second characters are upper case, we leave it alone.
+     * <p>
+     * Thus "FooBah" becomes "fooBah" and "X" becomes "x", but "URL" stays
+     * as "URL".
+     *
+     * Taken from
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/base/AccessorKey.java]: Taken from what?

File: drools-core/src/main/java/org/drools/core/base/AccessorKey.java [hunk 3]
```diff
@@ -109,4 +108,32 @@ public String toString() {
...
+     *
+     * @param  name The string to be decapitalized.
+     * @return  The decapitalized version of the string.
+     */
+    public static String decapitalizeFieldName(String name) {
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/base/AccessorKey.java]: I'm not sure here is a good fit for this method, what about [`StringUtils`](https://github.com/kiegroup/drools/blob/main/drools-core/src/main/java/org/drools/core/util/StringUtils.java)?

File: drools-core/src/main/java/org/drools/core/definitions/InternalKnowledgePackage.java [hunk 4]
```diff
@@ -144,15 +148,30 @@
...
+
+    default Class<?> getFieldType(Class<?> clazz, String leftValue) { return null; }
+
+    default InternalReadAccessor getReader(String className, String fieldName, AcceptsReadAccessor target) { return null; }
+
+    default Collection<KnowledgeBuilderResult> getWiringResults(Class<?> classType, String fieldName) { return Collections.emptyList(); }
+
+    default InternalReadAccessor getFieldExtractor( TypeDeclaration type, String timestampField, Class<?> returnType ) { return null; }
+
+    default void setClassFieldAccessorCache(ClassFieldAccessorCache classFieldAccessorCache) { }
```
Comment (danielezonca) [inline drools-core/src/main/java/org/drools/core/definitions/InternalKnowledgePackage.java:176]: Are all "empty" methods?
Are they used only by MVEL?

File: drools-traits/src/main/resources/META-INF/kie/drools-traits/kie.conf (hunk 1 above)
Comment (mariofusco) [reply drools-traits/src/main/resources/META-INF/kie/drools-traits/kie.conf:4]: Now there is also a `MVELCoreComponentFactory`. If there are only mvel it should use that one, if there are mvel and traits it should use this and that's why it has higher precedence.

File: drools-core/src/main/java/org/drools/core/definitions/InternalKnowledgePackage.java (hunk 4 above)
Comment (mariofusco) [reply drools-core/src/main/java/org/drools/core/definitions/InternalKnowledgePackage.java:176]: Yes.
